homeassistant/components/screenlogic/services.py

```diff
@@ -0,0 +1,100 @@
+"""Services for ScreenLogic integration."""
+
+import logging
+
+from screenlogicpy import ScreenLogicError
+import voluptuous as vol
+
+from homeassistant.core import ServiceCall, callback
+from homeassistant.exceptions import HomeAssistantError
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.service import async_extract_config_entry_ids
+from homeassistant.helpers.typing import HomeAssistantType
+
+from .const import (
+    ATTR_COLOR_MODE,
+    DOMAIN,
+    SERVICE_SET_COLOR_MODE,
+    SUPPORTED_COLOR_MODES,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+INTEGRATION_SERVICES = [SERVICE_SET_COLOR_MODE]
+
+SET_COLOR_MODE_SCHEMA = cv.make_entity_service_schema(
+    {
+        vol.Required(ATTR_COLOR_MODE): vol.In(SUPPORTED_COLOR_MODES),
+    },
+)
+
+
+@callback
+def async_load_screenlogic_services(hass: HomeAssistantType):
+    """Set up services for the ScreenLogic integration."""
+    existing_services = hass.services.async_services().get(DOMAIN)
+    if existing_services and any(
+        service in INTEGRATION_SERVICES for service in existing_services
+    ):
+        # Integration-level services have already been added. Return.
+        return
+
+    async def extract_screenlogic_config_entry_ids(service_call: ServiceCall):
+        screenlogic_entry_ids = []
+        for entry_id in await async_extract_config_entry_ids(hass, service_call):
```

Comment:
`async_extract_config_entry_ids` returns a set so there cannot be duplicates

Reply:
Ah yes. Thanks.